lib/polymorphic_integer_type/belongs_to_polymorphic_association_extension.rb

```diff
@@ -3,7 +3,7 @@ module Associations
     class BelongsToPolymorphicAssociation < BelongsToAssociation
       private def replace_keys(record)
         super
-        owner[reflection.foreign_type] = record.class
+        owner[reflection.foreign_type] = record.class.base_class
```

Comment:
Was getting a lot of failures in Themis when this was set to record.class. I'm a bit unsure about what is the correct answer here. My limited understanding is:

- if we use `#base_class` it will fail to find mappings where the type mapping uses a child class.
- if we use `#class` it will fail to find mappings where the type mapping uses a parent class

base_class passes CI on 5.1.7 at least (haven't been able to run full test suite under 5.2 yet)

@doliveirakn any insights?